libs/sdk-core/src/invoice.rs

```diff
@@ -189,8 +195,17 @@ pub fn add_lsp_routing_hints(
     Ok(invoice_builder.build_raw()?)
 }
 
+pub fn validate_network(invoice: Invoice, network: Network) -> InvoiceResult<()> {
+    match invoice.network() == network.into() {
+        true => Ok(()),
+        false => Err(InvoiceError::InvalidNetwork(anyhow!(
+            "Invoice network does not match config"
+        ))),
+    }
+}
+
 /// Parse a BOLT11 payment request and return a structure contains the parsed fields.
-pub fn parse_invoice(bolt11: &str) -> InvoiceResult<LNInvoice> {
+pub fn parse_invoice(bolt11: &str, network: Option<Network>) -> InvoiceResult<LNInvoice> {
```

Comment:
This change makes the binding interface different than the rust interface (we added one parameter only to the rust layer).
I am not sure we should overload the upper layer validation down to this function. Perhaps we should validate on the caller instead?